### PR TITLE
[Blog Post] How to Build Original Projects

### PR DESCRIPTION
**Pull request by reyes-dev**
Hello everyone!

I wrote a blogpost about project building. It's basically complete but could use a looking over, and I have a few minor concerns regarding the content I'd like to share
- I used the term "original projects" to refer to basically any custom, homegrown project. I wonder if this is the right term and will convey the correct meaning?
- In a section on planning I describe how I would take a root idea and begin planning out each phase based on core features. I used an example of a hypothetical project but I wonder if my example was too contrived or not general enough for everyone to understand?

Also open on suggestions to adding or trimming content.

### CODE REVIEW

**Comment by crutchcorn** [comment]
Hey hey @reyes-dev - I know we spoke in DMs about adding a blog post to the site. However, we can't merge this PR as-is. We don't add folks to the `unicorns.json` file unless we're also merging other work from them at the same time.

If you add your blog post to this PR, we can then review and merge ASAP

File: content/blog/building-original-projects/index.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
+---
+{
+  title: "How to Build Original Projects",
+  description: "People often say that you should build your own projects to grow as a developer, without explaining how. Let's learn how to actually plan and build your own custom projects.",
+  published: '2023-10-31',
+  authors: ["jarrodreyes"],
```
Comment (crutchcorn) [inline content/blog/building-original-projects/index.md:6]:
```suggestion
  authors: ["reyes-dev"],
```

File: content/blog/building-original-projects/error.png
Comment (crutchcorn) [inline content/blog/building-original-projects/error.png:1]: I'd remove this file and replace it with a code block. It:

- Improves our SEO
- Improves A11Y
- Avoids blurry screenshots on hi-rez screens

File: content/blog/building-original-projects/index.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+- It is a new creation of your mind; there is no guide or tutorial that can teach you directly how to build it
+- It solves a real world problem you or others have
+
+To clarify, an original project is not necessarily a completely new product that nobody has ever thought of, or invented before (although it can be!).  I also don't mean that you are *not allowed* to reference guides, tutorials, documentation, other people's code or Google in the process of building an original project. The opposite is true, and I highly encourage using all resources at your disposal.
+
+Rather, the essence of the original project is autonomy and personal vision. You are laying all the groundwork: the ideas, plan, research, and features. Then, you will execute on your plan of action by translating the broad ideas into code. By the end, you will have in your hands new software for your use case. You will be trading in the prepackaged, tutorial-project approach for a slightly scarier, but ultimately more rewarding approach, driven mainly by your creativity and problem-solving skills.
```
Comment (crutchcorn) [inline content/blog/building-original-projects/index.md:20]: Guidance for future blog posts - smaller paragraphs. We've found that - at least with technical writing - they're:

- Easier to read at a glance
- Enhance user retention in reading

Try to stay to 2-3 sentences (~300-350 characters) at most.

File: content/blog/building-original-projects/index.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+
+# Why should you build original projects?
+There can be a myriad of motivations for pursuing an original project. These can range from learning new technologies/concepts, to making that dream app you wish existed, wishing a plugin or tool existed for some piece of software you already use, or even just fun and entertainment. Whatever your reasons, building an original project is sure to be a powerful learning experience that challenges you in new ways. This act of creation will draw out all your latent abilities, giving you a fresh view into your strengths and weaknesses as an engineer. You will level up, build something useful and have fun along the way.
```
Comment (crutchcorn) [inline content/blog/building-original-projects/index.md:23]: > making that dream app you wish existed

Link your project! It'll provide context and promote your work!

File: content/blog/building-original-projects/index.md [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+- Displaying data: Users visiting the site should be able to see recently posted articles upon landing on your site.
+- API access: You need to grab data from Steam and Twitch, and display it on the page.
+- User interface: How should the layout of the website look visually? You might need some mockups. 
+
+Each of these can be further broken down into subtasks, until you end up with something like:
+-  Implementing Blogposts
+	1. First create simple text-only posts that can be posted by authorized users and publicly viewed by anyone. Make some pages with forms for creating new posts.
+	2. Look into how to allow authors to upload/insert video files to their blog post.
+	3. Find a way to link Steam account data to author account, and display relevant information such as the amount of hours the author has played this game.
+	4. With Twitch API access in hand, display past streams of the game at the bottom of any given blogpost.
```
Comment (crutchcorn) [inline content/blog/building-original-projects/index.md:63]: While I love the idea of a list here - as it stands it feels incomplete:

image

To fix this, I would:

- Add more root (non-numbered) bullet-points
- Make the list match the `Authorization` and `Blog posts` lists from above. Parity is key here for glanceability
- Don't be afraid of nested list items
  > Something like this nested block quote might help add visual diversity

File: content/blog/building-original-projects/index.md [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+
+```javascript
+const sum = add(a, b);
+```
+
+Now we can follow the stack trace up further:
+
+```at add (script.js:5:3)
+at add (script.js:5:3)
+```
```
Comment (crutchcorn) [inline content/blog/building-original-projects/index.md:193]: [Important] This breaks out code embed. To fix this:

````suggestion
```
at add (script.js:5:3)
```
````

File: content/blog/building-original-projects/index.md [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+
+Which tells us that we should look on line 5:
+
+```
```
Comment (crutchcorn) [inline content/blog/building-original-projects/index.md:197]:
````suggestion
```javascript
````

File: content/blog/building-original-projects/index.md [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+function add(a, b) {
+  return c; // This looks like the culprit
+}
+```
+
+Bingo! It seems like this line, `return c;` is the issue, and true to the error's message, it is trying to reference a `c` variable that has not been defined in the scope of this function. What we wanted to do was store the addition of `a` and `b` in a `c` variable and return it, like so:
+
+```
```
Comment (crutchcorn) [inline content/blog/building-original-projects/index.md:205]:
````suggestion
```javascript
````

File: content/blog/building-original-projects/index.md [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+
+> When to use: If your problem stems from a bug that produces an error message, always!
+
+## Try Debugging
+How you debug will depend on your choice of programming language and code editor, and there are even debuggers that work on the command line or in your browser. If you use Chrome, one way to practice is to use the built-in debugger, [as outlined in this official guide to debugging JavaScript in Chrome](https://developer.chrome.com/docs/devtools/javascript/). The basic premise of debuggers are that you can observe the behavior of your code as it's executing. By setting one or several `breakpoints`, your debugger will allow you to slowly step forward line-by-line starting from one of the breakpoints set. The benefit of this is that you can examine the state of your variables in the current scope as you move through the execution of your code.
```
Comment (crutchcorn) [inline content/blog/building-original-projects/index.md:221]: Did you know that you can also use Chrome's debugger in Node? Maybe we could add a link to my article? :) 

https://unicorn-utterances.com/posts/debugging-nodejs-programs-using-chrome

File: content/blog/building-original-projects/index.md [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+2. When the form for a new post gets submitted, send the post through the websocket
+3. Listen for when the post is received by the server
+4. When a post is received, run a function that appends the new post to an element
+
+### Divide and conquer
+ A sufficiently complex problem will benefit from breaking it down to the smallest subproblems, eventually hitting bedrock and reaching tasks that you can actually take action on. Implementing WebSockets sounds daunting at first. But with our instructions, we just need to tackle one tiny subproblem at a time, starting with the first. The first subproblem can be solved with a single line of code:
+
+`let socket = new WebSocket(url);`
+
+> When to use:  Whenever you're lost or confused, in need of clarity and a path to action.
```
Comment (crutchcorn) [inline content/blog/building-original-projects/index.md:122]: **Love** this usage of `When to use`

File: content/blog/building-original-projects/index.md [hunk 10]
```diff
@@ -0,0 +1,285 @@
...
+- The FAQs. Oftentimes, common problems will be addressed in good docs in the FAQs section. Though not all docs will have a FAQ, if they do, check it.
+- Showcase/Examples. Consists of coded implementations of the technology, this is usually where I go if I've read the main content and glanced at the API.
+
+> When to use: You feel that you're missing something, or can't understand how to use a technology for desired effect, it's likely your knowledge is has shaky foundations.
+
+## Seek the help of tech communities
+If you've entered some communities, introduce yourself and try leveraging their help in a respectful manner for new perspectives into your issue! Find the relevant channel or subforum of your community, and then communicate your problem clearly, providing relevant context, errors, and code samples. Sometimes the mere act of explaining your problem to others will grant new insight on how to solve it.
+
+Please be polite towards others when asking for help. Programming-related problems can be frustrating and at times emotional, but if people are offering to help you out of kindness, be mindful of your reaction to their advice.
```
Comment (crutchcorn) [inline content/blog/building-original-projects/index.md:241]: @AlexChadwickP wrote an awesome article for our site about this:

https://unicorn-utterances.com/posts/how-to-ask-good-questions

Maybe worth a link?

**Review by crutchcorn — CHANGES_REQUESTED** [review]
@reyes-dev... 👏👏👏 Well done, my friend. Sincerely, as your first article this is a standout piece. It's in-depth, well articulated, and has a tone that's wonderfully you.

Please keep writing, even if not for UU - I'm eager to read more 🤘✨

I've left some feedback, but these are primarily nit-picks. We could ship as-is today and it'd be a great launch.

File: content/blog/building-original-projects/index.md (hunk 4 above)
Comment (reyes-dev) [reply content/blog/building-original-projects/index.md]: I updated the section, was that closer to what you had in mind?

Comment (crutchcorn) [reply content/blog/building-original-projects/index.md]: LGTM!

File: content/blog/building-original-projects/index.md (hunk 1 above)
Comment (crutchcorn) [reply content/blog/building-original-projects/index.md:6]:
```suggestion
  authors: ["reyes-dev"],
```

File: content/blog/building-original-projects/index.md [hunk 11]
```diff
@@ -0,0 +1,367 @@
+---
+{
+  title: "How to Build Original Projects",
+  description: "People often say that you should build your own projects to grow as a developer, without explaining how. Let's learn how to actually plan and build your own custom projects.",
+  published: '2023-10-31',
+  authors: ["jarrodreyes"],
```
Comment (crutchcorn) [inline content/blog/building-original-projects/index.md:6]:
```suggestion
  authors: ["reyes-dev"],
```

File: content/blog/building-original-projects/index.md [hunk 12]
```diff
@@ -0,0 +1,367 @@
+---
+{
+  title: "How to Build Original Projects",
+  description: "People often say that you should build your own projects to grow as a developer, without explaining how. Let's learn how to actually plan and build your own custom projects.",
+  published: '2023-10-31',
```
Comment (crutchcorn) [inline content/blog/building-original-projects/index.md:5]:
```suggestion
  published: '2023-11-09',
```

**Review by crutchcorn — APPROVED** [review]

**Pull request merged**